lib/msgraph/client.go

```diff
@@ -207,8 +255,22 @@ func (c *Client) request(ctx context.Context, method string, uri string, header
 			return resp, nil
 		}
 
-		graphError, err := readError(resp.Body)
-		resp.Body.Close()
+		respBody, err := utils.ReadAtMost(resp.Body, teleport.MaxHTTPResponseSize)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		if err := resp.Body.Close(); err != nil {
+			c.logger.WarnContext(req.Context(), "Failed to close http.Responde body", "error", err)
+		}
+
+		c.logger.DebugContext(req.Context(), "Request failed",
+			"body", string(respBody),
+			"status", resp.StatusCode,
+			"url", req.URL,
+			"client_request_id", requestID,
+		)
```

Comment:
What error are we logging here? 
Request error is returned on line 250 https://github.com/gravitational/teleport/pull/58661/files#diff-1ac4f0db366e973f5e9d8da8329769522668c43ee44688aef0f9f5ba5864a8f1R250

Response body read error is retuedn in line 259?

Reply:
This logs responses that don't fall within the range of `resp.StatusCode >= 200 && resp.StatusCode < 400` from line 254.

Reply:
Oh got it. I think I just read the "Request failed" message and thought we are handling non-existent `err` field here.